src/Libraries/Microsoft.Extensions.AI/ChatCompletion/DistributedCachingChatClient.cs

```diff
@@ -95,10 +104,155 @@ protected override async Task WriteCacheStreamingAsync(string key, IReadOnlyList
     /// <summary>Computes a cache key for the specified values.</summary>
     /// <param name="values">The values to inform the key.</param>
     /// <returns>The computed key.</returns>
-    /// <remarks>The <paramref name="values"/> are serialized to JSON using <see cref="JsonSerializerOptions"/> in order to compute the key.</remarks>
-    protected override string GetCacheKey(params ReadOnlySpan<object?> values)
+    /// <remarks>
+    /// <para>
+    /// The <paramref name="values"/> are serialized to JSON using <see cref="JsonSerializerOptions"/> in order to compute the key.
+    /// </para>
+    /// <para>
+    /// The generated cache key is not guaranteed to be stable across releases of the library.
+    /// </para>
+    /// </remarks>
+    protected override string GetCacheKey(params ReadOnlySpan<object?> values) =>
+        GetDefaultCacheKey(values, _jsonSerializerOptions);
```

Comment:
Would there be any value in making this virtual instance method `public` as well? That would provide a path for people to get the correct cache key even if the logic is overridden by a subclass.

I know it's not as convenient as the static method since it forces the developer to instantiate (or have access to an existing instance of) the DCCC. But at least it would be possible.

Reply:
> Would there be any value in making this virtual instance method public as well? That would provide a path for people to get the correct cache key even if the logic is overridden by a subclass.

Do you have an example for where that would be useful? If you're looking to emulate the exact behavior of a derived type in order to get the same cache keys that instance is using, you not only need to have the algorithm, but you also need to know what inputs are being passed to the method, and that's an implementation detail.

I'm not against it, I'm just unclear what it unlocks.

Reply:
I don't have an end-to-end scenario in mind, so I'm glad you brought this up.

My comment assumed that whatever end-to-end scenario applies for the static logic would apply to per-type logic too. But now I'm wondering if the use case is different.

Are you saying the use case for the static logic isn't to anticipate the cache keys that we'd be using, but rather as a more general-purpose "hash a load of MEAI objects for reasons unrelated to DCCC"? If that's the case then arguably the API shouldn't be on DCCC.

Reply:
> Are you saying the use case for the static logic isn't to anticipate the cache keys that we'd be using, but rather as a more general-purpose "hash a load of MEAI objects for reasons unrelated to DCCC"?

Yes, albeit still related to caching, just not via DCCC. Exposing it here seemed like a reasonable middle ground between exposing such that it _could_ be used for other things even if not promoted for that. But we could also move it to AIJsonUtilities if we want it to be an obviously broader thing.

Reply:
Meh, I'll just move it. It's just a hash for hashing json serialized anything.

Reply:
Moved it to be AIJsonUtilities.HashDataToString. ("HashData" matches the name used on our crypto types, and I included the "ToString" suffix because a) the crypto HashData produces bytes, and b) we might want a byte-based version in the future as well.)